surefire-api/src/main/java/org/apache/maven/surefire/api/report/LegacyPojoStackTraceWriter.java

```diff
@@ -81,10 +79,15 @@ public String smartTrimmedStackTrace()
         result.append( "#" );
         result.append( testMethod );
         SafeThrowable throwable = getThrowable();
+        final String excClassName = throwable.getTarget().getClass().getName();
         if ( throwable.getTarget() instanceof AssertionError )
+        if ( throwable.getTarget() instanceof AssertionError
```

Comment:
This `if ( ... )` is a pure trojan horse.

Reply:
This section of code is suited for another Jira ticket.
The way we split the fix purpose we can better separate commits per issue in the git history.

Reply:
unintentional. I'll amend the PR

Reply:
Feel free to open a new Jira and PR with acceptance of asserions exceptions.

Reply:
@sman-81 
I am not sure why the author of `smartTrimmedStackTrace()` ignores other exceptions. Try to open a new PR and try to turn the IF statement to:
`if ( throwable.getTarget() != null )`
I want to know if the old code makes strong sense to the tests.

Reply:
The exception message may have multiple lines. They may produce a huge report on the console. My advice is to print only the first line if exists.